tests/test_publish.py

```diff
@@ -0,0 +1,41 @@
+from pathlib import Path
+from uuid import uuid4
+
+from unittest.mock import Mock, patch
+
+from faculty_mill import publish
+
+
+def test_that_get_report_by_name_calls_the_right_endpoint():
+
+    mock_report = Mock()
+    mock_report.id = "test id"
+    mock_report.name = "test name"
+    mock_client = Mock()
+    mock_client.list.return_value = [mock_report]
+
+    with patch(
+        "sherlockml.client", return_value=mock_client
+    ) as mock_client_creator:
+
+        result = publish.get_report_id_by_name("test name", "project id")
+        mock_client_creator.assert_called_once_with("report")
+        mock_client.list.assert_called_once_with("project id")
+        assert result == mock_report.id
+
+
+def test_that_publish_calls_client_method_correctly_with_all_ids_set():
+
+    mock_client = Mock()
```

Comment:
I suggest using `pytest-mock` as we do in the sherlockml library. This provides a pytest fixture that unapplies the mock automatically after the test, avoiding the need to indent most of the content of tests in context managers.

Have a look at the sherlockml library tests for examples of how to use it.

Reply:
I decided to go for the decorator option instead, as that seems a bit more obvious and avoids extra dependencies / fixture magics

Reply:
Well it's nice as you can then write your mocks as pytest fixtures and share them. Pytest fixtures are unpleasantly magic for Python, but they save a lot of boilerplate compared to standard unittest constructs.

There is arguably also value in some level of consistency in our published Python code - we use pytest-mock in most Python code in the SherlockML team, both internally and in the sherlockml library.

Reply:
Yeah - I use it on superintendent, as well, but it feels icky :P If we get more tests that need this in the future, we can always implement it in a separate PR.